app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -224,8 +224,10 @@ def process_os(instance)
         }
       end
 
+      # find both OS and SKU if possible, otherwise just the OS type.
       def guest_os(instance)
-        image_reference = instance.properties.storage_profile.image_reference
+        image_reference = instance.properties.storage_profile.try(:image_reference)
+        return instance.properties.storage_profile.os_disk.os_type unless image_reference
         image_reference.offer + " " + image_reference.sku.tr('-', ' ')
```

Comment:
Looks good.  I know this isn't part of your change, but can you change this `+` to use interpolation instead, since you are already in this code.